rust/worker/src/storage/s3.rs

```diff
@@ -21,11 +21,17 @@ use aws_sdk_s3::error::ProvideErrorMetadata;
 use aws_sdk_s3::error::SdkError;
 use aws_sdk_s3::operation::create_bucket::CreateBucketError;
 use aws_sdk_s3::primitives::ByteStream;
+use aws_sdk_s3::types::CompletedMultipartUpload;
+use aws_sdk_s3::types::CompletedPart;
+use aws_smithy_types::byte_stream::Length;
 use futures::Stream;
 use std::clone::Clone;
 use std::time::Duration;
 use thiserror::Error;
 
+// todo: make this more principled
+const MULTIPART_UPLOAD_CHUNK_SIZE: u64 = 1024 * 1024 * 32;
```

Comment:
need some guidance on what to set this to; from what I read S3 will not re-chunk the file after it's finished uploading--this is the size that the parts get internally stored as on AWS
implication: [AWS recommends](https://docs.aws.amazon.com/whitepapers/latest/s3-optimizing-performance-best-practices/use-byte-range-fetches.html) that the chunk size when fetching in parallel be the same size as the original multipart upload chunk size

Reply:
4MB is our standard blocksize but we should also make this part of the config

Reply:
e0872cae

Reply:
Ah got it, do we want to just not do multipart if < min part size?

Reply:
we could, not sure if it's worth forking the logic for now if this isn't a bottleneck

Reply:
yeah its fine